src/main/java/prgms/vouchermanagementapp/controller/CommandExecutor.java

```diff
@@ -0,0 +1,92 @@
+package prgms.vouchermanagementapp.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import prgms.vouchermanagementapp.io.CommandType;
+import prgms.vouchermanagementapp.io.IOManager;
+import prgms.vouchermanagementapp.model.Amount;
+import prgms.vouchermanagementapp.model.Ratio;
+import prgms.vouchermanagementapp.voucher.VoucherManager;
+import prgms.vouchermanagementapp.voucher.VoucherType;
+import prgms.vouchermanagementapp.voucher.model.Voucher;
+
+import java.text.MessageFormat;
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class CommandExecutor {
+
+    private final IOManager ioManager;
+    private final VoucherManager voucherManager;
+    private final RunningState runningState;
+
+    @Autowired
+    public CommandExecutor(IOManager ioManager, VoucherManager voucherManager) {
+        this.ioManager = ioManager;
+        this.voucherManager = voucherManager;
+        this.runningState = new RunningState();
+    }
+
+    public void run() {
+        while (runningState.isRunning()) {
+            Optional<CommandType> commandType = ioManager.askCommand();
+            commandType.ifPresent(this::executeCommand);
```

Comment:
`ioManager.askCommand()`가 Optional을 반환하고 있어 NPE에 대해 고려하지 않아도 되겠군요 👍 
그렇다면 무의미한 지역 변수를 사용하지 않고 

```suggestion
            ioManager.askCommand()
            		.ifPresent(this::executeCommand);
```
이렇게 작성해보는 것은 어떨까요?

Reply:
[Refactor(command): 커맨드 입력 예외처리 로직 변경](https://github.com/prgrms-be-devcourse/springboot-basic/commit/e760ad99892b68bb35fa913890a68d7ad710bd6d)
[Refactor(command): 커맨드 입력을 요청하고 실행하는 로직을 간략화](https://github.com/prgrms-be-devcourse/springboot-basic/commit/4078b608a1dc0fb6ef16c43cb0cb0253096e3550)

 `한 줄에 점을 하나만 찍는다`를 지키고, 코드 한 줄에 하나의 작업만 이루어지는 것이 가독성 면에서 더 좋겠다고 생각하여 의도적으로 개행처리를 하였습니다. 그런데 추천해주신 suggestion을 보니 오히려 이 쪽이 더 자연스럽게 읽히는 것 같아 suggestion처럼 리팩터링하였습니다!
 
또한 `한 줄에 점을 하나만 찍는다` 를 지키는 것과 관련된 `디미터의 법칙`을 찾아보고, 이것이 다른 객체가 가지고 있는 값을 직접참조하는 것을 방지하고 `메시지를 통해 객체가 협력`하기 위한 방법이라고 이해하였습니다. 따라서, 경우에 따라 점을 하나 이상 찍을 수도 있음을 알게되었습니다❗️